### PR TITLE
Follow-up PR for updating `ci-base-image`

### LINKED ISSUE #1896: Follow-up PR for updating `ci-base-image`
- [x] All GitHub workflows should be updated to use `nightly-2024-02-04` and `stable`
- [x] `RUST_TOOLCHAIN` should be updated to `stable` once a new `ci-base-image` is published
- [ ] `Cargo.lock` may be updated by `cargo update`

### PR DESCRIPTION
**Pull request by mattheworris**
# Goal
The goal of this PR is:
- [x] All GitHub workflows should be updated to use `nightly-2024-02-04` and `stable`
- [x] `RUST_TOOLCHAIN` should be updated to stable once a new `ci-base-image` is published
- [x] `Cargo.lock` may be updated by `cargo update`

Closes #1896

# How to Test
- 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -323,7 +323,7 @@ jobs:
           srtool --version
       - name: Build Deterministic WASM
         run: |
-          RUST_LOG=debug SRTOOL_TAG="1.70.0" srtool build \
+          RUST_LOG=debug SRTOOL_TAG="1.75.0" srtool build \
```
Comment (saraswatpuneet) [inline .github/workflows/release.yml:326]: should we do 1.76 here?

Comment (mattheworris) [reply .github/workflows/release.yml:326]: 1.75.0 appears to be the latest version.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -361,8 +361,8 @@ jobs:
         uses: actions/configure-pages@v4
       - name: Build Docs
         run: |
-          rustup target add wasm32-unknown-unknown --toolchain nightly-2023-07-13
-          RUSTDOCFLAGS="--enable-index-page -Zunstable-options" cargo +nightly-2023-07-13 doc --no-deps --workspace --features frequency
+          rustup target add wasm32-unknown-unknown --toolchain nightly-2024-02-04
+          RUSTDOCFLAGS="--enable-index-page -Zunstable-options" cargo +nightly-2024-02-04 doc --no-deps --workspace --features frequency
```
[the remark this answers is not in the document]
Comment (mattheworris) [reply .github/workflows/release.yml:365]: We are using `stable` for the builds. We are only overriding for a couple actions, e.g. rust docs, lint, that require unstable features.

We want `.rust-toolchain.toml` to be `stable` for everything that doesn't need unstable features.

To my knowledge, we can only specify one `channel` at a time in `.rust-toolchain.toml`.

**Comment by p5150j** [comment]
@mattheworris ⚠️ warnings for docs and code coverage are fine

**Review by saraswatpuneet — APPROVED** [review]

**Review by p5150j — APPROVED** [review]

**Review by wilwade — APPROVED** [review]
If it passes CI, ✅

**Pull request merged**